File: 0387/main.py
from typing import Dict


class Solution:
    def firstUniqChar(self, s: str) -> int:
        hashmap: Dict[str, int] = {}
        index: int = 0
        value: str = ""
        for index, value in enumerate(s):
            if hashmap.get(value) != None:
                hashmap[value] = -1
            else:
                hashmap[value] = index

        minIndex: int = 1e9
        for (value, index) in hashmap.items():
            if index != -1 and index < minIndex:
                minIndex = index
        
        if minIndex == 1e9:
            return -1
        return minIndex

File: 0387/main.swift
class Solution {
    func firstUniqChar(_ s: String) -> Int {
        var hashmap = [Character: Int]()

        for (index, value) in s.enumerated(){
            if let _ = hashmap[value]{
                hashmap[value] = -1
            } else {
                hashmap[value] = index
            }
        }

        var minIndex: Int = Int.max
        for (_, index) in hashmap{
            if index != -1 && index < minIndex{
                minIndex = index
            }
        }
        if minIndex == Int.max{
            return -1
        }
        return minIndex
    }
}